service/chartconfig/v1/appr/appr.go

```diff
@@ -10,31 +18,97 @@ import (
 type Config struct {
 	Logger micrologger.Logger
 
-	Address string
+	Address      string
+	Organization string
 }
 
 // Client knows how to talk with a CNR server.
 type Client struct {
-	logger micrologger.Logger
+	httpClient *http.Client
+	logger     micrologger.Logger
 
-	address string
+	base *url.URL
 }
 
 // New creates a new configured appr client.
 func New(config Config) (*Client, error) {
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
 	}
-
 	if config.Address == "" {
 		return nil, microerror.Maskf(invalidConfigError, "address must not be empty")
 	}
+	if config.Organization == "" {
+		return nil, microerror.Maskf(invalidConfigError, "organization must not be empty")
+	}
+
+	// set client timeout to prevent leakages.
+	hc := &http.Client{
+		Timeout: time.Second * 5,
```

Comment:
Does it make sense to have a constant or a flag for the timeout?

Reply:
Totally, added a constant for now.